### PR TITLE
[Jsonifier] to v0.9.97 and [DiscordCoreAPI] to v2.0.7

### LINKED ISSUE #45: Incompatible dependencies - jsonifier v0.9.96
Compiler: MSVC 19.38
OS: Windows 11
Command: `vcpkg install discordcoreapi`

Built failed with `jsonifier` v0.9.96 which introduced many changes, like:

```log
Include\discordcoreapi/Utilities/Etf.hpp(612): error C2039: 'float_t': is not a member of 'jsonifier::concepts'
jsonifier/TypeEntities.hpp(68): note: see declaration of 'jsonifier::concepts'
...
alloc.deallocate(static_cast<object_type*>(objectValue), 1);
```

Could you please help to take a look? Thanks.

### PR DESCRIPTION
**Pull request by RealTimeChris**
Jsonifier Changes:
* Added likely/unlikely to a bunch of if/else statements.
* Swapped std::copy_n for std::memcpy in a few places.
* Moved a couple masks for the Neon functions out into global scope.
* Replaced jsonifier_internal::const_iterator with jsonifier_internal::iterator.
* Moved the vector of prettifier states out of the function scope to the prettifier classes' scope.
* Separated jsonifier_internal::string_literal out into its own header.
* Moved a bunch of string-utility functions out into the derailleur class.
* Replaced the join function with combineLiterals.
* Modified the logic of the parse_impl classes.
* Implemented a compile-time jump table to improve parsing performance significantly.
* Replaced the float/double parsing/serializing methods with more conformant version.
* Modified the serializing code to no longer rely on hash-maps for serializing member.
* Marked a bunch of additional functions as noexcept.
* Added an ALWAYS_INLINE macro in addition to the INLINE one.
* Modified the CPU-architecture selection method.

DiscordCoreAPI Changes:
* Updated to keep in track with Jsonifier dependency.
* Added a detect-cpu-architecture script.

Also to fix [this](https://github.com/RealTimeChris/DiscordCoreAPI/issues/45) issue 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
> * Added a detect-cpu-architecture script.

When does this script run? There is already the "target triplet". "Native" optimizations would need to be enabled by custom triplet.

**Comment by RealTimeChris** [comment]
> > * Added a detect-cpu-architecture script.
> 
> When does this script run? There is already the "target triplet". "Native" optimizations would need to be enabled by custom triplet.
It runs during cmake's configuration step.

**Comment by Cheney-W** [comment]
x64-uwp failures:
```
D:\b\discordcoreapi\src\v2.0.7-699923b831.clean\Include\discordcoreapi/JsonSpecializations.hpp(79): error C2993: 'derived_type': is not a valid type for non-type template parameter '__formal'
D:\b\discordcoreapi\src\v2.0.7-699923b831.clean\Include\discordcoreapi/JsonSpecializations.hpp(90): error C3855: 'jsonifier_internal::parse_impl': template parameter '__formal' is incompatible with the declaration
D:\b\discordcoreapi\src\v2.0.7-699923b831.clean\Include\discordcoreapi/JsonSpecializations.hpp(90): error C3855: 'jsonifier_internal::parse_impl': template parameter 'value_type' is incompatible with the declaration
D:\b\discordcoreapi\src\v2.0.7-699923b831.clean\Include\discordcoreapi/JsonSpecializations.hpp(90): error C3855: 'jsonifier_internal::parse_impl': template parameter 'iterator' is incompatible with the declaration
D:\b\discordcoreapi\src\v2.0.7-699923b831.clean\Include\discordcoreapi/JsonSpecializations.hpp(115): error C3855: 'jsonifier_internal::parse_impl': template parameter '__formal' is incompatible with the declaration
D:\b\discordcoreapi\src\v2.0.7-699923b831.clean\Include\discordcoreapi/JsonSpecializations.hpp(115): error C3855: 'jsonifier_internal::parse_impl': template parameter 'value_type' is incompatible with the declaration
D:\b\discordcoreapi\src\v2.0.7-699923b831.clean\Include\discordcoreapi/JsonSpecializations.hpp(115): error C3855: 'jsonifier_internal::parse_impl': template parameter 'iterator' is incompatible with the declaration
```
https://dev.azure.com/vcpkg/public/_build/results?buildId=106787&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by dg0yt** [comment]
> > * Added a detect-cpu-architecture script.
> > 
> > When does this script run? There is already the "target triplet". "Native" optimizations would need to be enabled by custom triplet.
> 
> It runs during cmake's configuration step. https://github.com/RealTimeChris/DiscordCoreAPI/blob/main/CMake/DCADetectArchitecture.cmake

IMO the port should make sure this is not used by default. Do not limit the portability of the binary artifacts beyond what is controlled by the target triplet.

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -305,11 +305,6 @@ discord-rpc:arm64-uwp=fail
 discord-rpc:x64-uwp=fail
 discordcoreapi:x64-linux=skip   # cause linux CI to hang https://github.com/microsoft/vcpkg/pull/40229
```
Comment (WangWeiLin-MV) [inline scripts/ci.baseline.txt:306]:
```suggestion
```

**Review by Cheney-W — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**